CHANGES.md

```diff
@@ -1,6 +1,473 @@
+* fixed a build issue on systems with low-resolution timestamps
+
+Launchers
+---------
+* added a `smp` launcher for simulating multi-locales on a shared-memory system
+  (see https://chapel-lang.org/docs/1.17/usingchapel/launcher.html#currently-supported-launchers)
+* added a `gasnetrun_ofi` launcher for running with GASNet over its OFI conduit
+  (see https://chapel-lang.org/docs/1.17/usingchapel/launcher.html#currently-supported-launchers)
+
+Compiler Performance
+--------------------
+* significantly improved the speed of --llvm compilation
+
+Third-Party Software Changes
+----------------------------
+* made Chapel default to using LLVM 6.0.0 when enabled
+  (see https://www.chapel-lang.org/docs/1.17/usingchapel/chplenv.html)
+* removed support for LLVM versions older than 4.x
+  (see https://www.chapel-lang.org/docs/1.17/usingchapel/chplenv.html)
+* updated hwloc to 1.11.9
+* updated fltk to 1.3.4-2
+
+Testing System
+--------------
+* dramatically improved the usability of the web-based performance graphs
+* added tab-completion support for start_test
+* made `start_test` process files, subdirectories in sorted order
+* allow performance testing flags to be overridden
+* added support for executable compopts files with conditionally empty output
+* added support for a -futures-only flag to parallel testing scripts
+* improved testing system's ability to run from a non-CHPL_HOME util/ directory
+* improved test system checks and support for valid valgrind configurations
+* fixed a race in code that limits the number of concurrently executing tests
+* fixed a sporadic error in tests using the high-precision timer option
+* expanded availability of FileCheck testing tool to CHPL_LLVM=system builds
+
+Developer-oriented changes: Configuration changes
+-------------------------------------------------
+* Added `CHPL_RUNTIME_ARCH` to specify architecture runtime is built for
+* Added `CHPL_TARGET_BACKEND_ARCH` to specify cpu-type used for `-march` flag
+* removed a special case and detected 64-bit ARM atomics the same as others
+
+Developer-oriented changes: Module changes
+------------------------------------------
+* converted most internal/standard/package module objects to use `[de]init()`
+* simplified how automatic `use`s are handled in internal modules
+
+Developer-oriented changes: Makefile improvements
+-------------------------------------------------
+* added support for computing the compiler version when using git worktree
+* added an error for unsupported flex versions when generating the lexer
+* fixed a problem preventing `make check -j` from succeeding
+
+Developer-oriented changes: Compiler Flags
+------------------------------------------
+* added a `--lifetime-checking` flag to enable prototype use-after-free checks
+* added a `--warn-unstable` flag to warn of changing language features
+* added --llvm-print-ir to produce an LLVM module that works with other tools
```

Comment:
--llvm-print-ir already existed (but I think we made it developer only). I might say `updated --llvm-print-ir ...`